trie/node.go

```diff
@@ -118,13 +120,14 @@ func computeAndSetNodeHash(n node) ([]byte, error) {
 func getNodeFromDBAndDecode(n []byte, db common.DBWriteCacher, marshalizer marshal.Marshalizer, hasher hashing.Hasher) (node, error) {
 	encChild, err := db.Get(n)
 	if err != nil {
+		log.Trace(common.GetNodeFromDBErrorString, "error", err, "key", n, "stack trace", string(debug.Stack()))
+
 		dbWithID, ok := db.(dbWriteCacherWithIdentifier)
 		if !ok {
-			log.Trace(common.GetNodeFromDBErrorString, "error", err, "key", n, "db type", fmt.Sprintf("%T", db))
+			log.Warn("db does not have an identifier", "db type", fmt.Sprintf("%T", db))
```

Comment:
log warn while returning the error. Why not wrap?

Reply:
Maybe use log trace if you already return this error in the caller side as well.

Reply:
Wrapped the err.